aiida/cmdline/commands/cmd_daemon.py

```diff
@@ -64,13 +74,17 @@ def start(foreground):
 
     print_client_response_status(response)
 
+    if number > 1:
```

Comment:
is there a reason to do it this way?
i.e. why not make the number of workers a parameter of `get_daemon_client` and just start with the right number of workers?

Reply:
I couldn't do this, because we do not actually start the daemon through the client. That just serves to query it once it is running and to stop it. The daemon is started through the hidden internal command `verdi daemon start-circus`.

However, I agree that the method I chose is sub-optimal so I changed it. I now pass `number` through to the sub command that we call, and there in the `arbiter` configuration, I launch the number of processes we want straight away.

I added an exception being raised if more than one processes are requested while running in the foreground, since that doesn't make any sense and added some more tests for this